tools/sof-ipc-timer.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2025 Intel Corporation. All rights reserved.
+
+'''The sof-ipc-timer collects module initialization and configuration
+timings from 'journalctl -k -o short-precise' output if SOF IPC debug
+is enabled.
+
+'''
+
+import re
+import pathlib
+import argparse
+from datetime import datetime
+
+class Component:
+    '''SOF audio component storage class'''
+    ppln_id: str
+    comp_id: int
+    wname: str
+
+    def __init__(self, ppln_id, comp_id, wname):
+        self.ppln_id = ppln_id
+        self.comp_id = comp_id
+        self.wname = wname
+
+    def __str__(self) -> str:
+        return f'{self.ppln_id}-{self.comp_id:#08x}'
+
+def read_comp_data(f):
+    '''Process the dmesg to get the component ID to component name mapping,
+    they are acquired from the line that contains 'Create widget':
+        [   59.622645] snd_sof:sof_ipc4_widget_setup: sof-audio-pci-intel-mtl 0000:00:1f.3:
+        Create widget host-copier.0.capture (pipe 3) - ID 4, instance 3, core 0"
+
+    By design in the kernel, pipeline ID is the instance ID of pipeline
+    widget, so it is acquired from the line that contains 'Create pipeline':
+        [   59.622134] snd_sof:sof_ipc4_widget_setup: sof-audio-pci-intel-mtl 0000:00:1f.3:
+        Create pipeline pipeline.3 (pipe 3) - instance 3, core 0
+
+    In practice, sof-test only capture kernel message and firmware trace generated during a test
+    case run. Mostly in manual tests, if the kernel message file contains multiple firmware runs
+    with overlapping information, the last one wins.
+    '''
+    comp_data = {}
+    ppln_id = None
+    for line in f:
+        if match_obj := re.search(r"Create (widget|pipeline)", line):
+            span_end_pos = match_obj.span()[1]
+            line_split = line[span_end_pos + 1:].split()
+            widget_name = line_split[0]
+            # In the linux kernel, IDA is used to allocated pipeline widget instance ID,
+            # this ID later is used for pipeline creation, thus becomes pipeline ID in the
+            # firmware. Note that ppln_id variable will be assigned properly at pipeline widget
+            # creation, because it is always the first one to be created before all other
+            # widgets in the same pipeline and pipelines are created sequentially.
+            if widget_name.startswith('pipeline'):
+                # remove ending comma(,) with [:-1]
+                ppln_id = int(line_split[5][:-1])
```

Comment:
To remove all leading and trailing commas whether there are some or not:
```suggestion
                ppln_id = int(line_split[5].strip(','))
```
This is also more readable, now you can drop the comment.